File: frontend/dto/baseSongAlbumResponse.ts
import { z } from "zod";
import { BaseArtistResponseSchema } from "@/dto";

export const BaseSongAlbumResponseSchema = z.object({
    provider: z.string(),
    publicId: z.string(),
    name: z.string(),
    artists: z.array(z.lazy(() => BaseArtistResponseSchema)),
    releaseDate: z.string(),
    internalImageUrl: z.string(),
});

export type BaseSongAlbumResponse = z.infer<typeof BaseSongAlbumResponseSchema>;

File: backend/core/responses/baseSongAlbumResponse.py
from typing import Sequence

from pydantic import BaseModel

from backend.core.responses.baseArtistResponse import BaseArtistResponse


class BaseSongAlbumResponse(BaseModel):
    provider: str
    publicId: str
    name: str
    artists: Sequence[BaseArtistResponse]
    releaseDate: str
    internalImageUrl: str
